### ISSUE TITLE
The code is different from the original paper

### ISSUE THREAD

**Issue by lmisssunl** [heykeetae/Self-Attention-GAN#54]
Hello, first of all thank you for your code, let me have a deeper study of SAGAN, but after reading your code I have the following questions:
1. In the original text, the relationship between β j, i and h(x) is N∑(i=1) βj, ih(xi), which is not the result of multiplying the torch.bmm matrix in your code
2. The third 1*1 convolution in the article (i.e. h(x)), the dimension obtained is not C, but C//8
3. The last layer in the original text has a 1 * 1 convolution, but it is not reflected in your code (maybe because of the second item I just mentioned, you changed the output C//8 to C, the last layer of 1 * 1 convolution is not ignored)

**Comment by valillon** [comment]
Yep, also found those discrepancies!

**Comment by valillon** [comment]
Apparently, as mentioned [here](https://github.com/ajbrock/BigGAN-PyTorch/issues/88), max pooling inside the attention layer is just motivated by design-wise to save computation/memory overhead. This should close the issue.

**Comment by WangDingok** [comment]
Actually, softmax(dim=-1) applies softmax over keys for a given query. dim=1 would normalize over queries for a given key, which is the opposite direc
